kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/metrics/MessageMetrics.java

```diff
@@ -69,4 +83,32 @@ private void count(Object msg, MeterProvider<Counter> counterProvider) {
         }
     }
 
+    private void incrementSize(Object msg, MeterProvider<DistributionSummary> distributionProvider) {
+        if (msg instanceof Frame frame) {
+            var apiKey = ApiKeys.forId(frame.apiKeyId());
+            short version = frame.apiVersion();
+            boolean decoded = frame.isDecoded();
+
+            final int size;
+            // TODO fix me!
```

Comment:
Something needs to be done here.

Reply:
Yeah, sorry, I only noticed after reviewing that it was still in draft. :man_facepalming: